hls-template/build_prj.tcl

```diff
@@ -16,8 +16,8 @@ foreach arg $::argv {
 
 open_project -reset myproject_prj
 set_top myproject
-add_files firmware/myproject.cpp -cflags "-I[file normalize nnet_utils]"
-add_files -tb myproject_test.cpp -cflags "-I[file normalize nnet_utils]"
+add_files firmware/myproject.cpp -cflags "-I[file normalize nnet_utils] -std=c++0x"
```

Comment:
Since we now rely on C++11, we should be using `-std=c++11`. c++0x is not recommended.

Reply:
My understanding was that Vivado HLS does not support -std=c++11, but c++0x is okay [1]. Maybe this is no longer true in most recent versions?

[1] https://forums.xilinx.com/t5/SDSoC-Environment-and-reVISION/Error-when-using-flag-quot-std-c-11-quot/td-p/704847

Reply:
I tried it on 2017.2, and it works.

Reply:
For CSIM or CSYNTH?

Reply:
csim. Didn't try synth.

Reply:
sorry if i missed it, is there a specific this flag is needed in the specific PR?

Reply:
I was wrong, it appears c++11 is not really supported and we should use c++0x [1].

[1] https://forums.xilinx.com/t5/Vivado-High-Level-Synthesis-HLS/Can-HLS-2017-02-support-std-c-11/td-p/896120

Reply:
Hmm... on the machine where we have Vivado 2017.2, gcc is 4.8.5 and it accepts `-std=c++11`. But the default one (4.6.3) does not. It appears we have manually changed the compiler at some point :-). Anyway, no change is needed.

@nhanvtran `constexpr` is C++11, used in `nnet_helpers.h`